### PR TITLE
Refactor Probability Map Producer Handler

### LINKED ISSUE #4381: Refactor ProbMapProducer
With the updates is WSI readers and WSI dataset, `ProbMapProducer` handler needs to be refactored to work properly. Also it would be great to make it thread safe.

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #4381 

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_wsireader_new.py [hunk 1]
```diff
@@ -121,109 +121,125 @@ class WSIReaderTests:
...
-            if self.backend == "cucim" and (len(img_expected.shape) < 3 or img_expected.shape[2] == 1):
-                # Until cuCIM addresses https://github.com/rapidsai/cucim/issues/230
-                return
-            reader = WSIReader(self.backend)
-            file_path = os.path.join(os.path.dirname(__file__), "testing_data", "temp_tiff_image_gray.tiff")
-            imwrite(file_path, img_expected, shape=img_expected.shape)
-            with self.assertRaises((RuntimeError, ValueError, openslide.OpenSlideError if has_osl else ValueError)):
-                with reader.read(file_path) as img_obj:
-                    reader.get_data(img_obj)
+        # @parameterized.expand([TEST_CASE_0])
```
Comment (Nic-Ma) [inline tests/test_wsireader_new.py:124]: May I know why comment all these tests?

Thanks.

File: monai/apps/pathology/handlers/prob_map_producer.py [hunk 2]
```diff
@@ -25,6 +25,10 @@
     Engine, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Engine")
 
 
+@deprecated(
+    since="0.8",
+    msg_suffix="use `monai.handler.ProbMapProducer` (with `monai.data.wsi_dataset.SlidingPatchWSIDataset`) instead.",
```
Comment (Nic-Ma) [inline monai/apps/pathology/handlers/prob_map_producer.py:30]: I don't quite understand why it's replaced by `monai.data.wsi_dataset.SlidingPatchWSIDataset`?

Thanks.

File: tests/test_wsireader_new.py (hunk 1 above)
Comment (drbeh) [reply tests/test_wsireader_new.py]: Accidentally! 😅

File: monai/apps/pathology/handlers/prob_map_producer.py (hunk 2 above)
Comment (drbeh) [reply monai/apps/pathology/handlers/prob_map_producer.py:30]: `ProbMapProducer` uses meta information that `SlidingPatchPatchReader` provides to create the probability maps. There has been some changes in those meta information when refactoring pathology datasets.

`monai.apps.handler.ProbMapProducer` <-> `monai.apps.data.MasedInferencePatchWSIDataset`
`monai.handler.ProbMapProducer` <-> `monai.data.wsi_dataset.SlidingPatchWSIDataset`

Comment (drbeh) [reply monai/apps/pathology/handlers/prob_map_producer.py:30]: @Nic-Ma is this clear?
Also I have added the keys to enum so they will be reused in any related dataset.

**Comment by drbeh** [comment]
@Nic-Ma, any other comments here? Otherwise, it should be ready to merge. Thanks

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update.
Looks good to me except for minor comments inline.

Thanks.

**Pull request merged**